### PR TITLE
Enable/Disable pages in wxToolbook

### PR DESCRIPTION
**Pull request by sz5000**
Functions to enable and disable pages inside toolbooks. Disabled icons are created through ConvertToDisabled(). With this functionality you can present disabled icons so that the user can expect more functionality and you do not need to add/remove pages in different states.

### CODE REVIEW

File: src/generic/toolbkg.cpp [hunk 1]
```diff
@@ -336,6 +338,25 @@ bool wxToolbook::DeleteAllPages()
     return wxBookCtrlBase::DeleteAllPages();
 }
 
+bool wxToolbook::EnablePage(size_t page, bool enable)
+{
+    GetToolBar()->EnableTool(page + 1, enable);
+    if (!enable && GetSelection() == (int)page) {
```
Comment (vadz) [inline src/generic/toolbkg.cpp]: Minor style nitpick: could you please move the opening brace to its own line, to conform to wx coding style?

File: src/generic/toolbkg.cpp [hunk 2]
```diff
@@ -336,6 +338,25 @@ bool wxToolbook::DeleteAllPages()
...
+        AdvanceSelection();
+    }
+    return true;
+}
+
+bool wxToolbook::EnablePage(wxWindow *page, bool enable)
+{
+    int pageIndex = FindPage(page);
+    if (pageIndex == wxNOT_FOUND) {
```
Comment (vadz) [inline src/generic/toolbkg.cpp]: Same style comment as above.

**Review by vadz — DISMISSED** [review]
This looks like a useful addition, thank you!

However the new functions really need to be documented in `interface/wx/tookbook.h`. Please add the new functions there (and please use `@since 3.1.2` in their documentation to indicate that they're new) and it's probably worth mentioning their existence in the class comment there. Ideal would also be to update the notebook sample (`samples/notebook`) to show this functionality in action, but this is not required for merging this PR, unlike the docs.

TIA!

**Comment by sz5000** [comment]
Comments are added to interface/wx/tookbook.h

**Comment by sz5000** [comment]
Seems that The Travis CI build test failed due to code not affected by this patch. Can the test be executed again?

**Pull request closed without merging**